pkgs/by-name/di/dim/package.nix

```diff
@@ -0,0 +1,100 @@
+{
+  lib,
+  stdenv,
+  rustPlatform,
+  fetchFromGitHub,
+  buildNpmPackage,
+  makeWrapper,
+  ffmpeg_5,
+  git,
+  pkg-config,
+  sqlite,
+  libvaSupport ? stdenv.hostPlatform.isLinux,
+  libva,
+}:
+rustPlatform.buildRustPackage rec {
+  pname = "dim";
+  version = "0-unstable-2023-12-29";
+  src = fetchFromGitHub {
+    owner = "Dusk-Labs";
+    repo = "dim";
+    rev = "3ccb4ab05fc1d7dbd4ebbba9ff2de0ecc9139b27";
+    hash = "sha256-1mgbrDnIkIdWy78uj4EjjgwBQxw/rIS1LCFNscXXPbk=";
+  };
+
+  frontend = buildNpmPackage {
+    pname = "dim-ui";
+    inherit version;
+    src = "${src}/ui";
+
+    postPatch = ''
+      ln -s ${./package-lock.json} package-lock.json
```

Comment:
Can we ask upstream to update their `yarn.lock` and use `fetchYarnDeps` instead?

Reply:
Or update it on the postFetch if they don't look much collaborative.

https://github.com/NixOS/nixpkgs/pull/261139